native-libs/test/main.cpp

```diff
@@ -47,8 +48,109 @@ void test_decode() {
     delete[] input;
 }
 
+void test_decompress() {
+    std::ifstream file("test/blocks.json.gzip", std::ios::binary | std::ios::ate);
```

Comment:
What happens initially when this file does not yet exist?

Reply:
this file is ignored by git. In case this file does not exist, a fast fail should happen.